hollow/src/main/java/com/netflix/hollow/core/write/objectmapper/HollowObjectTypeMapper.java

```diff
@@ -849,6 +895,224 @@ private Object parseBoxedWrapper(FlatRecordReader reader) {
             return null;
         }
 
+        private Object parseBoxedWrapper(FlatRecordTraversalObjectNode record) {
+            switch (fieldType) {
+                case BOOLEAN:
+                    return record.getFieldValueBooleanBoxed(fieldName);
+                case INT:
+                    int intValue = record.getFieldValueInt(fieldName);
+                    if (intValue == Integer.MIN_VALUE) {
+                        return null;
+                    }
+                    return Integer.valueOf(intValue);
+                case SHORT:
+                    int shortValue = record.getFieldValueIntBoxed(fieldName);
```

Comment:
ask: cannot use the boxed helper here
```suggestion
                    int shortValue = record.getFieldValueInt(fieldName);
```